File: todotasks-React/src/View/TaskList.js
import React from 'react'
import TaskItem from './TaskItem'
import Task from '../Model/Task'

class TaskList extends React.Component {
    constructor(props) {
        super(props)
        this.tasks = [
            new Task("1", false, false),
            new Task("2", false, false),
            new Task("3", false, false),
            new Task("4", false, false),
            new Task("5", false, false),
            new Task("6", false, false)
        ]
    }
    render() {
        return (
            <div>
                {
                    this.tasks.map(task => {
                        return <TaskItem task={task} key={task.id}></TaskItem>
                    })
                }
            </div>
        )
    }
}

export default TaskList

File: ToDoTasks-Flutter/lib/View/TaskList.dart
import 'package:flutter/material.dart';
import 'package:todotasks/Model/Task.dart';
import 'TaskItem.dart';

class TaskList extends StatefulWidget {
  @override
  _TaskList createState() => _TaskList();
}

class _TaskList extends State<TaskList> {
  List<Task> tasks = [
    Task(
        content: "",
        isDone: false,
        isStarred: false,
        creationTime: DateTime.now()),
    Task(
        content: "",
        isDone: false,
        isStarred: false,
        creationTime: DateTime.now()),
    Task(
        content: "",
        isDone: false,
        isStarred: false,
        creationTime: DateTime.now()),
    Task(
        content: "",
        isDone: false,
        isStarred: false,
        creationTime: DateTime.now()),
    Task(
        content: "",
        isDone: false,
        isStarred: false,
        creationTime: DateTime.now())
  ];

  @override
  Widget build(BuildContext context) {
    return ListView.builder(
        itemCount: tasks.length,
        itemExtent: 60,
        itemBuilder: (context, index) {
          return TaskItem(task: tasks[index]);
        });
  }
}
